packages/protocol/contracts/common/UniswapFeeHandlerSeller.sol

```diff
@@ -0,0 +1,194 @@
+pragma solidity ^0.5.13;
+
+import "../common/interfaces/IFeeHandlerSeller.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "../stability/interfaces/IExchange.sol";
+import "../stability/interfaces/ISortedOracles.sol";
+import "./UsingRegistry.sol";
+import "../stability/StableToken.sol";
+import "../common/FixidityLib.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "../common/Initializable.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/math/Math.sol";
+import "./FeeHandlerSeller.sol";
+
+import "../uniswap/interfaces/IUniswapV2RouterMin.sol";
+import "../uniswap/interfaces/IUniswapV2FactoryMin.sol";
+
+contract UniswapFeeHandlerSeller is IFeeHandlerSeller, FeeHandlerSeller {
+  using SafeMath for uint256;
+  using FixidityLib for FixidityLib.Fraction;
+
+  mapping(address => address[]) public routerAddresses;
+  uint256 constant MAX_TIMESTAMP_BLOCK_EXCHANGE = 20;
+
+  event ReceivedQuote(address router, uint256 quote);
```

Comment:
Can we extend it also with which token this quote is for ? Also this could be indexed